bitswap/client/client.go

```diff
@@ -139,7 +167,16 @@ func New(parent context.Context, network bsnet.BitSwapNetwork, bstore blockstore
 	sim := bssim.New()
 	bpm := bsbpm.New()
 	pm := bspm.New(ctx, peerQueueFactory, network.Self())
-	pqm := bspqm.New(ctx, network)
+	if bs.pqm == nil { // not set with the options
+		// network can do dialing and also content routing.
+		pqm, err := rpqm.New(ctx, network, network, rpqm.WithMaxProviders(10))
```

Comment:
If we don't want to pass-in a Host and a ContentRouter directly to the Client, we need to rely on Network, which has them, but it would be better to remove Content Routing capabilities from the Network because it is just wrapping the content Router and nothing else at this point.